app/scripts/modules/core/src/search/infrastructure/RecentlyViewedItems.tsx

```diff
@@ -9,12 +9,23 @@ import { ReactInjector } from 'core/reactShims';
 
 import { ISearchResult, ISearchResultPodData, SearchResultPods } from './SearchResultPods';
 
+export interface IChildComponentProps {
+  results: ISearchResultPodData[];
+  onRemoveItem?: (categoryName: string, itemId: string) => void;
+  onRemoveProject?: (projectId: string) => void;
+}
+
+export interface IRecentlyViewedItemsProps {
+  Component: React.ComponentClass<IChildComponentProps> | React.StatelessComponent<IChildComponentProps>;
```

Comment:
Why `Component` over `render`?

Reply:
I tend to prefer using a `Component` prop that gets rendered via `<Component/>` syntax rather than a `render` prop that gets called as a function via `{render(props)}` syntax. Main reason is that it means you can succinctly pass in components on their own  rather than always wrapping things in a function, e.g. `<RecentlyViewedItems Component={SearchResultPods}/>`, while still retaining the ability to use a render function that becomes a stateless component in practice.

The reason I don't like wrapping components in functions all the time is because I've run into some mount/unmount/render thrashing with stateful components inside render functions due to React being unable to tell it should re use the old component instance. Naming the prop itself `Component` rather than `render` just falls out of the fact that `<render/>` looks weird 🙂.

Happy to go with an always-function API if we're averse to this pattern for whatever reason.

Reply:
I don't have a preference, but wanted to hear your reasoning.

We should probs use `React.ComponentType<IChildComponentProps>` for the type though.

Reply:
cool — I actually have it changed to `React.ComponentType` on my local follow-up branch after seeing you use it in another PR! Slowly but surely starting to get the hang of the day-to-day with TS